### PR TITLE
Make all tests compile with only mocks from mockall

### PR DESCRIPTION
**Pull request by AurelienFT**
## This PR include the big change : 

- Use mockall instead of custom mock with channels everywhere

### This PR improves : 

- The channels structure is now divided to pass only the broadcast channels when they are needed

## Next steps : 

- Make a global, unify, standard and easy to use test framework for the whole massa projects
- Add more checks using mockall on the methods of the interfaces called

### CODE REVIEW

File: massa-protocol-worker/src/tests/ban_nodes_scenarios.rs [hunk 1]
```diff
@@ -114,21 +125,11 @@ fn test_protocol_bans_node_sending_block_header_with_invalid_signature() {
...
-                consensus_event_receiver,
-                pool_event_receiver,
-                selector_event_receiver,
-            )
         },
     )
 }
 
 #[test]
-#[serial]
```
Comment (modship) [inline massa-protocol-worker/src/tests/ban_nodes_scenarios.rs:131]: removing "serial" does not cause any problem with other tests ?

File: massa-protocol-worker/src/tests/block_scenarios.rs [hunk 2]
```diff
@@ -637,31 +604,24 @@ fn test_protocol_sends_blocks_when_asked_for() {
...
-                pool_event_receiver,
-                selector_event_receiver,
-            )
         },
     )
 }
 
 #[test]
-#[serial]
+//TODO: fix
```
Comment (modship) [inline massa-protocol-worker/src/tests/block_scenarios.rs:623]: should be done ?

**Review by modship — DISMISSED** [review]
amazing job that greatly improves test management

File: massa-consensus-exports/src/lib.rs [hunk 3]
```diff
@@ -12,10 +12,13 @@ pub mod error;
 pub mod events;
 pub mod export_active_block;
 
-pub use channels::ConsensusChannels;
+pub use channels::{ConsensusBroadcasts, ConsensusChannels};
 pub use controller_trait::{ConsensusController, ConsensusManager};
 pub use settings::ConsensusConfig;
 
+#[cfg(feature = "testing")]
```
[the remark this answers is not in the document]
Comment (AurelienFT) [reply massa-consensus-exports/src/lib.rs:19]: As the mock will be only used in other modules and not himself there is no need to expose it with the test macro

File: massa-protocol-worker/src/tests/block_scenarios.rs (hunk 2 above)
Comment (AurelienFT) [reply massa-protocol-worker/src/tests/block_scenarios.rs]: Done now

File: massa-protocol-worker/src/tests/ban_nodes_scenarios.rs (hunk 1 above)
Comment (AurelienFT) [reply massa-protocol-worker/src/tests/ban_nodes_scenarios.rs:131]: Tested and don't seems too CI is agree with me so we can stay like this :)

File: massa-factory-worker/src/tests/scenarios.rs [hunk 4]
```diff
@@ -1,65 +1,56 @@
...
 }
 
-/// Creates a block with a roll buy operation in it.
-#[test]
-#[ignore]
-fn basic_creation_with_operation() {
-    let keypair = KeyPair::generate(0).unwrap();
-    let mut test_factory = TestFactory::new(&keypair);
+// /// Creates a block with a roll buy operation in it.
+// #[test]
```
[the remark this answers is not in the document]
Comment (AurelienFT) [reply massa-factory-worker/src/tests/scenarios.rs:12]: I want to let them I will make the tests of the factory in the next days.

File: massa-consensus-worker/src/tests/three_four_threads_scenarios.rs [hunk 5]
```diff
@@ -215,69 +243,92 @@ fn test_fts_multiple_max_cliques_1() {
...
+            register_block(&consensus_controller, block_2_2.clone(), storage.clone());
+
             // Should have 4 max cliques:
             // [block_1_0, block_1_1, block_1_2]
             // [block_1_1, block_1_2, block_2_0]
             // [block_1_2, block_2_0, block_2_1]
             // [block_2_0, block_2_1, block_2_2]
-            let mut status = tc
-                .consensus_controller
+            std::thread::sleep(Duration::from_millis(2000));
```
[the remark this answers is not in the document]
Comment (AurelienFT) [reply massa-consensus-worker/src/tests/three_four_threads_scenarios.rs]: I have recuded them at one t0 each period

File: massa-consensus-worker/src/tests/scenarios.rs [hunk 6]
```diff
@@ -278,24 +248,18 @@ fn test_parallel_incompatibility() {
...
                 while latest_extra_blocks.len() > cfg.delta_f0 as usize + 1 {
                     latest_extra_blocks.pop_front();
                 }
             }
-
             let latest_extra_blocks: HashSet<BlockId> = latest_extra_blocks.into_iter().collect();
             let status = consensus_controller
                 .get_block_graph_status(None, None)
                 .expect("could not get block graph status");
+            println!("AURELIEN: Test {:?}", status.best_parents);
```
[the remark this answers is not in the document]
Comment (AurelienFT) [reply massa-consensus-worker/src/tests/scenarios.rs]: Done

File: massa-factory-worker/src/tests/tools.rs [hunk 7]
```diff
@@ -38,9 +26,6 @@ use massa_wallet::test_exports::create_test_wallet;
 /// Then you can use the method `get_next_created_block` that will manage the answers from the mock to the factory depending on the parameters you gave.
```
Comment (Leo-Besancon) [inline massa-factory-worker/src/tests/tools.rs:26]: You removed the get_next_created_block method so remove it in doc comment here too

File: massa-grpc/src/tests/mock.rs [hunk 8]
```diff
@@ -36,192 +26,20 @@ use massa_versioning::versioning::{MipStatsConfig, MipStore};
...
 /// generate a grpc public service
 /// # Arguments
 /// * `addr` - the address to bind to
 /// # Returns
 /// * `MassaPublicGrpc` - the grpc public service
 #[allow(dead_code)]
 pub(crate) fn grpc_public_service(addr: &SocketAddr) -> MassaPublicGrpc {
-    let consensus_controller = MockConsensusControllerImpl::new();
-
+    let consensus_ctrl = Box::new(MockConsensusController::new());
```
Comment (Leo-Besancon) [inline massa-grpc/src/tests/mock.rs:36]: What's the goal of this deadcode function? Should we put it somewhere else now that "mock.rs" contains only that?

**Review by Leo-Besancon — COMMENTED** [review]
Made a quick pass over this PR, looks like good work :)

File: massa-grpc/src/tests/mock.rs (hunk 8 above)
Comment (AurelienFT) [reply massa-grpc/src/tests/mock.rs:35]: It comes from an other PR but it seems that a confusion was made between testing and test and so I fixed it which remove the utility of this.

File: massa-factory-worker/src/tests/tools.rs (hunk 7 above)
Comment (AurelienFT) [reply massa-factory-worker/src/tests/tools.rs]: Done.

**Review by modship — APPROVED** [review]

**Comment by AurelienFT** [comment]
bors merge

**Pull request merged**